tests/5.1/dispatch/test_dispatch_is_device_ptr.c

```diff
@@ -0,0 +1,74 @@
+//===---- test_dispatch_is_device_ptr.c -----------------------------------------===//
+// 
+// OpenMP API Version 5.1
+//
+// Uses dispatch construct as context for variant directive. Uses is_device_ptr
+// clause to ensure that the list item is in the device region.
+//
+// Inspired by "OpenMP 5.1 Features: The Dispatch Construct" video:
+// https://www.youtube.com/watch?v=ruugaX95gIs
+// 
+//===-------------------------------------------------------------------------------===//
+
+#include <omp.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <math.h>
+#include "ompvv.h"
+#include <stdbool.h>
+
+#define N 1024
+
+int errors;
+int i = 0;
+int *arr;
+
+void add_dev(int *arr);
+
+#pragma omp declare variant(add_dev) match(construct={dispatch}) 
+void add(int *arr){
+    #pragma omp parallel for
+    for (int i = 0; i < N; i++){ // Base function adds 2 to array values
+        arr[i] = arr[i]+2;
+    }
+}
+
+void add_dev(int *arr){
+    #pragma omp target
+    for (int i = 0; i < N; i++){
+        arr[i] = arr[i]+4; // Variant function adds 4 to array values
+    }
+}
+
+int test_wrapper() { 
+    int t;
+    errors = 0;
+    t = omp_get_default_device();
+    arr = (int *)omp_target_alloc( sizeof(int)*N, t);
+    #pragma omp target
+    {
+        #pragma omp parallel for
+        for(int i = 0; i < N; i++){
+            arr[i] = i;
+        }
+    }
+    #pragma omp dispatch is_device_ptr(arr)
```

Comment:
refer to page 54 line 6-7; we should accept both variant functions, use infomsg to print which version was used